models/__init__.py

```diff
@@ -0,0 +1,35 @@
+from .densenet import *
+from .resnet import *
+from .wresnet import *
+
+
+def get_model(model_type, num_classes=10, dataset_name="cifar10", **kwargs):
+
+    ## NOTE: the unfortunate use of lambda here can be dropped as soon as
+    ## PEP 622 https://www.python.org/dev/peps/pep-0622/ is released
+    ## and replaced with pattern matching
+    model_zoo = {
+        # ResNets
+        "resnet18": lambda kwargs: ResNet18(num_classes=num_classes, variant=dataset_name, **kwargs),
```

Comment:
Copying from slack for any future reference:

From @jack-willturner 

it's my fault - i wrote it like that

I acknowledge it's horrible but it can very soon be removed once python 3.10 is released
the idea is that you have if you have a higher order dictionary:
```d = {
  'one': One(a,b,c)
  'two': Two(d,e)
}```
this is evaluated eagerly (i.e. when you declare the dictionary, both One(a,b,c) and Two(d,e) are evaluated
so even if you only do d['one'] then both One(a,b,c) and Two(d,e) will be evaluated
if we have kwargs then we might not have access to d,e for example
the hacky fix is to pass any extra kwargs through a lambda. that way python evaluates all the lambdas and only passes the args through when you call them
if you can't read that then it's  too hacky. i'll rewrite it today
I just hate the idea of having a long if-else